### PR TITLE
Enhancement: Dev-9453 World Map No Data Scenario

### PR DESCRIPTION
**Pull request by Matt-DiPaolo**
## Enhancement: Dev-9453 World Map No Data Scenario
https://websupport.cdc.gov/browse/DEV-9453

When there is no data to show in the map, a "No Data" message is added to the legend and a tooltip is created with no data on hover.

## Testing Steps

Scenario: Upload [dev-9453-config.json](https://github.com/user-attachments/files/17816638/dev-9453-config.json) and navigate to the dashboard preview.

Expected: There is a world map with some countries highlighted according to the Legend and the Legend has a category call "No Data"

Hover over a country that is gray.
Expected: A tooltip appears with the country name and "No Data" at the bottom of the tooltip.

## Self Review

- I have added testing steps for reviewers
- My changes generate no new warnings

### CODE REVIEW

File: packages/map/src/CdcMap.tsx [hunk 1]
```diff
@@ -831,6 +831,20 @@ const CdcMap = ({
...
+    if (state.general.geoType === 'world') {
+      const runtimeDataKeys = Object.keys(runtimeData)
+      const isCountriesWithNoDataState =
+        obj.data === undefined ? false : !countryKeys.every(countryKey => runtimeDataKeys.includes(countryKey))
+
+      if (result.length > 0 && isCountriesWithNoDataState) {
+        result.push({
+          min: null,
+          max: null,
+          color: '#E6E6E6'
```
Comment (adamdoe) [inline packages/map/src/CdcMap.tsx:843]: The default background color has changed to --coolGray10. You should be able to get that using the helper function `getGeoFillColor` found in `/map/helpers/colors.ts`

File: packages/map/src/components/WorldMap/WorldMap.tsx [hunk 2]
```diff
@@ -98,7 +103,8 @@ const WorldMap = () => {
         legendColors = applyLegendToRow(geoData)
       }
 
-      const geoStrokeColor = state.general.geoBorderColor === 'darkGray' ? 'rgba(0, 0, 0, 0.2)' : 'rgba(255,255,255,0.7)'
+      const geoStrokeColor =
```
Comment (adamdoe) [inline packages/map/src/components/WorldMap/WorldMap.tsx:106]: Use the `getGeoStroke` color helper in `map/helpers/colors.ts` file

**Review by adamdoe — CHANGES_REQUESTED** [review]
This is looking good. Lets switch to using the color helpers in /helpers/colors.ts. I don't know if its needed across all map types quite yet, but I did notice that the example on the ticket was a state map. Do you want to add in all of the other variations now as well? Thanks!

File: packages/map/src/components/WorldMap/WorldMap.tsx [hunk 3]
```diff
@@ -104,8 +104,13 @@ const WorldMap = () => {
         legendColors = applyLegendToRow(geoData)
       }
 
+<<<<<<< HEAD
```
Comment (adamdoe) [inline packages/map/src/components/WorldMap/WorldMap.tsx:107]: Resolve merge conflicts

File: packages/map/src/components/WorldMap/WorldMap.tsx [hunk 4]
```diff
@@ -104,8 +104,13 @@ const WorldMap = () => {
...
       const geoStrokeColor = getGeoStrokeColor(state)
       const geoFillColor = getGeoFillColor(state)
+=======
+      const geoStrokeColor =
+        state.general.geoBorderColor === 'darkGray' ? 'rgba(0, 0, 0, 0.2)' : 'rgba(255,255,255,0.7)'
+>>>>>>> 20a2d8030 (Enhancement: Dev-9453 World Map No Data Scenario)
```
Comment (adamdoe) [inline packages/map/src/components/WorldMap/WorldMap.tsx:113]: Resolve

File: packages/map/src/components/WorldMap/WorldMap.tsx [hunk 5]
```diff
@@ -167,6 +171,11 @@ const WorldMap = () => {
           strokeWidth={strokeWidth}
           style={styles}
           path={path}
+<<<<<<< HEAD
```
Comment (adamdoe) [inline packages/map/src/components/WorldMap/WorldMap.tsx:174]: Resolve

**Review by adamdoe — CHANGES_REQUESTED** [review]

**Review by adamdoe — APPROVED** [review]

File: packages/map/src/CdcMap.tsx (hunk 1 above)
Comment (Matt-DiPaolo) [reply packages/map/src/CdcMap.tsx]: @adamdoe  I am trying to get this to work. I am not understanding how getGeoFillColor should work. I keep getting a black background.

Comment (adamdoe) [reply packages/map/src/CdcMap.tsx]: Hi @Matt-DiPaolo - Josh made changes earlier today. You can check his commit for the updates if you want to review them.

**Pull request merged**